### ISSUE TITLE
Build Curated Corpus of Academic Cover Pages from Platform Providers

### ISSUE THREAD

**Issue by donaldbraman** [donaldbraman/docling-testing#29]
# Issue #23: Build Curated Corpus of Academic Cover Pages from Platform Providers

## Summary
Create a curated, visually reviewable corpus of academic cover pages with the goal of identifying patterns in how platforms (HeinOnline, JSTOR, ProQuest, university servers, and publishers) prepend cover pages to PDFs. This corpus will serve as ground truth for identifying and extracting the `cover` class in our 7-class document segmentation model.

## Context
Initial analysis assumed the first page of each PDF is the cover page. However, academic platforms typically:
1. **Prepend platform cover pages** (e.g., HeinOnline header page, JSTOR watermark pages)
2. **Insert institutional/access pages** (university library metadata, access controls)
3. **Add archival metadata** (version history, provenance information)
4. **May not include true article cover** if document is multi-article collection

The "true" cover page (article title, authors, abstract) may be page 2, 3, or later.

Current situation:
- 427 PDFs collected from law review websites
- Unknown: What percentage have platform-prepended pages?
- Unknown: Cover page patterns by platform/source
- Unknown: How many true pages before article begins?

**Ref:** docs/multiclass-training.md - 10-class schema includes `Title`, `Author_Affiliation` as distinct classes

## Objective
1. Extract and visually review first 3 pages from all 427 PDFs
2. Identify cover page patterns by source/platform
3. Build CSV index: `pdf_file | page_number | is_platform_cover | is_article_cover | platform | notes`
4. Create visual browseable gallery for manual annotation
5. Determine optimal pages to sample for cover page training data

### 1. Cover Page Extraction and Indexing
- [ ] Extract first 3 pages from all 427 PDFs as PNG images:
  - Directory structure: `data/cover_pages_extracted/{journal_slug}/page_{1,2,3}.png`
  - DPI: 150 (for quality/filesize balance)
  - Size limit: <500KB per page
- [ ] Create index CSV: `data/cover_pages_index.csv`
  ```csv
  pdf_filename,page_1_type,page_2_type,page_3_type,detected_platform,true_article_start_page,source_journal,notes
  harvard_law_review_article1.pdf,platform_cover,article_cover,article_body,heinonline,2,"Harvard Law Review","HeinOnline header page",
  michigan_law_review_article2.pdf,article_cover,article_body,article_body,university_server,1,"Michigan Law Review","No platform cover",
  ```

### 2. Platform/Source Pattern Detection
- [ ] Analyze extracted pages to detect:
  - **Platform signatures** (HeinOnline branding, JSTOR watermark, ProQuest banner, university library footer)
  - **Common patterns** (e.g., "All HeinOnline PDFs start with 2-page header")
  - **Institutional markers** (university logo, access/no-access notices)
  - **Archival metadata** (version numbers, upload dates)
- [ ] Categorize each PDF:
  - Platform(s) detected (e.g., "HeinOnline", "JSTOR", "Direct from Publisher", "University Server", "Unknown")
  - Pages before actual article begins (1, 2, 3, 4+)
  - Confidence: High/Medium/Low
- [ ] Report statistics:
  - Distribution: X% have 1 platform page, Y% have 2 pages, Z% have none
  - By journal: Which journals have platform wrappers? (e.g., "80% of Texas Law PDFs have HeinOnline covers")
  - By platform: What's typical cover depth per platform?

### 3. Manual Visual Review and Annotation
- [ ] Create HTML gallery for visual review: `cover_pages_gallery.html`
  - Displays all first pages as image grid
  - Organized by journal
  - Shows detected platform/type
  - Sortable by: journal, platform, page_count
  - Modal popup for zoomed view
- [ ] Manual annotation task (team review, not just automated):
  - [ ] Review gallery and confirm/correct automated classification
  - [ ] For each PDF, decide: "What is the true first page of the article?"
  - [ ] Mark if page contains author names, title, or abstract (indicators of article start)
  - [ ] Note any unusual layouts
- [ ] Update index with manual annotations: `is_article_cover_verified`

### 4. Cover Page Type Classification
- [ ] Define clear categories for covers found:
  - `platform_header`: Publisher/platform branding (HeinOnline, JSTOR, etc.)
  - `institutional_access`: University library metadata/access page
  - `article_titlepage`: Article title + author names + affiliation (TRUE COVER)
  - `article_abstract`: Abstract or introduction text
  - `article_body`: Body text (article has begun)
  - `metadata_only`: Version info, archival timestamp, but no content
  - `unknown`: Could not determine
- [ ] For each type, record:
  - Count: how many PDFs have this type on page 1, 2, 3?
  - Example: "4 PDFs have platform_header on page 1"
  - Dominant type by journal

### 5. Cover Sample Selection
- [ ] Identify high-quality cover page samples for training:
  - Criteria: Clear `article_titlepage` with visible title + authors + affiliations
  - Minimum 20 per journal (where available)
  - Target: 500-1,500 cover samples across diverse journals
- [ ] Create curated subset: `data/cover_pages_training/`
  - Extract true cover pages (pages 2-4 of their PDFs) as PNG
  - Skip platform headers and access pages
  - Organize by journal
- [ ] CSV mapping: `training_pdf_file | source_journal | extracted_from_page | quality_rating`

## Success Metrics

1. **Extraction completeness**: 100% of 427 PDFs have pages 1-3 extracted
2. **Platform pattern identified**: ≥80% of PDFs correctly classified by platform type
3. **Manual review**: 90%+ agreement between automated detection and human review
4. **Training sample count**: ≥500 distinct, high-quality cover page images with verified labels
5. **Diversity**: ≥30 different journals represented in training samples

## Deliverables

1. **Cover Pages Index CSV** (`data/cover_pages_index.csv`)
   - 427 rows (one per PDF)
   - Columns: pdf_filename, page_1_type, page_2_type, page_3_type, detected_platform, true_article_start_page, source_journal, notes, manual_verification_status

2. **Visual Gallery** (`cover_pages_gallery.html`)
   - Browseable interface showing all first pages
   - Organized by journal
   - Platform detection displayed
   - Sortable/filterable
   - Zoom capability
   - Can be opened locally in browser

3. **Platform Analysis Report** (`docs/cover_page_platform_analysis.md`)
   - Summary of platform signatures found
   - Statistics: distribution of cover depths
   - By-journal breakdown (which have platform wrapping?)
   - Common patterns and outliers

4. **Curated Training Corpus** (`data/cover_pages_training/`)
   - 500-1,500 PNG files of verified article cover pages
   - Organized: `{journal_slug}/page_001.png`
   - Metadata CSV: source, quality, verification status

5. **Extraction Script** (`scripts/corpus_building/extract_and_index_cover_pages.py`)
   - Extracts first N pages from PDFs as PNG
   - Auto-detects platform signatures
   - Generates index and gallery
   - Reusable for future PDF collections

### Platform Signature Detection
Look for keywords/images:
- **HeinOnline**: "HeinOnline", copyright notice, volume/issue info at top
- **JSTOR**: "JSTOR" watermark, stable URL pattern, purchase info
- **ProQuest**: ProQuest logo, database name
- **arXiv**: arXiv ID, submission date, metadata box
- **Publisher direct**: DOI, journal copyright notice, no watermark
- **University server**: University logo, institutional repository notice, access policies

### Technical Approach
1. Use `pypdf` or `pdf2image` to extract pages as images
2. Use OCR or text extraction to find platform keywords
3. Analyze image signature (look for headers with logos, watermarks)
4. Rule-based classification, then HTML gallery for human review

### Quality Standards for Training
Only include cover pages that clearly show:
- Article title (primary indicator)
- Author name(s) (secondary indicator)
- Journal/publication info (tertiary indicator)
Skip pages that are:
- Platform headers with no article info
- Blank or nearly blank
- Only abstract text (may be redundant with body_text class)

## Testing

- [ ] All 427 PDFs successfully extract first 3 pages
- [ ] Gallery loads in browser without errors
- [ ] CSV has no blank critical fields
- [ ] Sample manual verification: 30 PDFs reviewed by eye, 90%+ match to automated detection
- [ ] Training corpus images are readable and properly labeled

## References

- docs/multiclass-training.md (10-class schema: `Title`, `Author_Affiliation`)
- HeinOnline format: https://heinonline.org/ (typical platform signature)
- JSTOR format: https://www.jstor.org/ (typical platform signature)
- pdf2image: https://pdf2image.readthedocs.io/

## Labels
`data-collection`, `corpus-building`, `cover-page-class`, `manual-review`, `priority:high`

## Milestone
Corpus Expansion for DoclingBERT v3 (Issue #21)

## Notes for Implementer

- This task benefits from visual review; don't skip manual verification
- Some PDFs may have unusual layouts (multi-column, vertical text) - flag these
- Consider image compression options to keep file sizes manageable
- Gallery can be generated in parallel with file extraction
